openssl_source/bypass_to_sgxssl.h

```diff
@@ -250,11 +250,6 @@ char * sgxssl___builtin___strcpy_chk(char *dest, const char *src, unsigned int d
 #define clock_gettime sgxssl_clock_gettime
 
 
-#define pthread_rwlock_init sgxssl_pthread_rwlock_init
-#define pthread_rwlock_rdlock sgxssl_pthread_rwlock_rdlock
-#define pthread_rwlock_wrlock sgxssl_pthread_rwlock_wrlock
-#define pthread_rwlock_unlock sgxssl_pthread_rwlock_unlock
-#define pthread_rwlock_destroy sgxssl_pthread_rwlock_destroy
 #define pthread_once sgxssl_pthread_once
```

Comment:
The Pthreads library included in the SGX SDK supports these APIs: pthread_once, pthread_key_create, pthread_setspecific, pthread_getspecific, pthread_key_delete, pthread_self, pthread_equal.
Wouldn't be cleaner to use those as well and remove the specific sgxssl_pthread functions?

Reply:
I agree, if the SDK already supports these functions, then it might be better to use them.  Let me compare the implementations just to make sure that there was not a specific reason why SGX SSL needed its own implementation.

Reply:
I did a quick review of the SDK implementation of the functions which are located in files here: [https://github.com/intel/linux-sgx/tree/master/sdk/pthread](https://github.com/intel/linux-sgx/tree/master/sdk/pthread).  They are implemented differently.  It is worth reconciling the SDK and intel-sgx-ssl; however, I think that it should be outside the scope of this PR.

Reply:
I agree, let's handle that change separately.